R/utils-list.R

```diff
@@ -34,28 +34,40 @@ strip_pts <- function(list, param){
   for (v in param){
     if (v %in% names(list) && !inherits(list[[v]], c('function','formula'))) {
       v.vals <- list[[v]]
-      out <- append(out, v.vals)
-      out.class <- ifelse(!all(is.na(v.vals)), class(v.vals), out.class)
+      out <- append_keepTZ(out, v.vals, tz=attr(v.vals, 'tzone'))
+      all.na <- all(is.na(v.vals))
+      out.class <- ifelse(!all.na, class(v.vals), out.class)
+      if(all.na){ 
+        class(out) <- out.class 
+      }
     } else {
       if (any(sapply(list, is.list))){
         u.list <- unname_c(list[sapply(list, is.list)])
         if(v %in% names(u.list)) {
           v.vals <- u.list[[v]]
-          out <- append(out, v.vals)
-          out.class <- ifelse(!all(is.na(v.vals)), class(v.vals), out.class)
+          out <- append_keepTZ(out, v.vals, tz=attr(v.vals, 'tzone'))
```

Comment:
Some keep the tz to be from base.vals, and some from append.vals, so I kept it outside. Most need to look at the tzone of the values you are appending because you could be appending to an empty object. However, sometimes you are appending an NA (see L66), and then you need to keep the tzone from the existing object.